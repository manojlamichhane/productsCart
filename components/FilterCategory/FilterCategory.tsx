"use client";
import { Combobox } from "@headlessui/react";
import React, { useState } from "react";
import { category } from "@/constants";

const FilterCategory = ({ selectedCategory }: { selectedCategory: string }) => {
  const [query, setQuery] = useState("");

  const filteredCategory =
    query === ""
      ? category
      : category.filter((item) => {
          return item.toLowerCase().includes(query.toLowerCase());
        });
  return (
    <Combobox value={selectedCategory} onChange={() => null}>
      <Combobox.Input
        className={"search-manufacturer__input"}
        onChange={(event) => setQuery(event.target.value)}
      />
      <Combobox.Options className={"search - manufacturer__options"}>
        {filteredCategory.map((item) => (
          <Combobox.Option
            key={item}
            value={item}
            className={"search-manufacturer__option"}
          >
            {item}
          </Combobox.Option>
        ))}
      </Combobox.Options>
    </Combobox>
  );
};

export default FilterCategory;
