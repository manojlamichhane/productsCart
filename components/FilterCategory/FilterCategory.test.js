import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import FilterCategory from "./FilterCategory";

describe("FilterCategory test", () => {
  test("Render check", () => {
    const { asFragment } = render(<FilterCategory />);
    expect(asFragment()).toMatchSnapshot();
  });
});
