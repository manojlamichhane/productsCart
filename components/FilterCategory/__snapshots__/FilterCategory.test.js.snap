// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterCategory test Render check 1`] = `
<DocumentFragment>
  <input
    aria-autocomplete="list"
    aria-expanded="false"
    class="search-manufacturer__input"
    data-headlessui-state=""
    id="headlessui-combobox-input-:r0:"
    role="combobox"
    type="text"
    value=""
  />
</DocumentFragment>
`;
